sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderRule.scala

```diff
@@ -30,4 +31,12 @@ trait CustomShuffleReaderRule extends Rule[SparkPlan] {
    * Returns the list of [[ShuffleOrigin]]s supported by this rule.
    */
   def supportedShuffleOrigins: Seq[ShuffleOrigin]
+
+  /**
+   * Whether it is used to coalesce partitions through AQE.
+   */
+  def isUsedToCoalescePartitions(s: ShuffleExchangeLike): Boolean = {
```

Comment:
This is too hacky. Let's add a new shuffle origin and include it in `OptimizeLocalShuffleReader.supportedShuffleOrigins`

Reply:
OK